rest/admin_api_test.go

```diff
@@ -1178,32 +1189,175 @@ func TestDBOfflineSingleResync(t *testing.T) {
 	require.NoError(t, base.JSONUnmarshal(response.Body.Bytes(), &body))
 	goassert.True(t, body["state"].(string) == "Offline")
 
-	input := bytes.NewBufferString("")
-	request, _ := http.NewRequest("POST", "http://localhost/db/_resync", input)
+	response = rt.SendAdminRequest("POST", "/db/_resync?action=start", "")
+	assertStatus(t, response, http.StatusOK)
+
+	// TODO: Should be fine but may need to add a wait method to slow down resync op
+	// Send a second _resync request.  This must return a 400 since the first one is blocked processing
+	assertStatus(t, rt.SendAdminRequest("POST", "/db/_resync?action=start", ""), 503)
+
+	err := rt.WaitForCondition(func() bool {
+		response := rt.SendAdminRequest("GET", "/db/_resync", "")
+		res := bytes.Compare([]byte(`{"status": "not running"}`), response.BodyBytes())
+		return res == 0
+	})
+	assert.NoError(t, err)
+
+	assert.Equal(t, int64(2000), rt.GetDatabase().DbStats.CBLReplicationPush().SyncFunctionCount.Value())
+}
+
+func TestResync(t *testing.T) {
+
+	testCases := []struct {
+		name               string
+		docsCreated        int
+		expectedSyncFnRuns int
```

Comment:
Is there anything in the tests to verify how many queries resync runs?  Should this test be checking the query stats and expecting a count of docsCreated/queryLimit + 1?

Reply:
Added check against this